.travis.yml

```diff
@@ -15,21 +15,13 @@ jobs:
         - make test
         - pip install safety
         - safety check || true
-    - name: “Python 3.8 make lint and selected pytests”
-      python: "3.8"
-      before_script: true  # override npm install below
-      script:
-        # On pull requests, run all flake8 tests on modified code
-        # if clause avoids `fatal: ambiguous argument 'origin/master'` on git push
-        - if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
-            make lint-diff;
```

Comment:
This is different; this runs the more aggressive flake8 against the diff, no? We still want that (unless our code has become compatible with the stricter flake8 config)

Reply:
A failure of `make lint-diff` does not currently fail the Travis CI test run.  There are two issues with the way that we are running `make lint-diff`:
1. [The Travis build passes even when `make lint-diff` fails.](https://travis-ci.org/github/internetarchive/openlibrary/jobs/720534297#L557)
2. [Flake8 is choking when handed binary data like .mrc files](https://travis-ci.org/github/internetarchive/openlibrary/jobs/719795714#L556)

I believe that these issues are __easy to fix__ but we would start failing the build when contributors fix typos.  I believe that we would be better served by [blackenIng our code](https://github.com/psf/black) before we make `make lint-diff` mandatory.  If not, we will frustrate our contributors with rework.  Manual rework may cause bugs.  If we do not blacken first then will spend a lot of code review time on educating contributors code formatting rules on parts of the lines that they do not wish to change.

Reply:
Yeah, let's enable this + make it mandatory. I don't think it'll be that annoying, since it'll only complain about the lines people are editing. 

To fix #2, would adding a `--filename=*.py` to scripts/flake8-diff.sh do the trick?

Reply:
I think we should consider blackening our code in the future, but not sure this is the best time considering how many things we have on our plate 😅